pkg/clusteragent/admission/mutate/autoinstrumentation/auto_instrumentation.go

```diff
@@ -329,35 +349,94 @@ func (w *Webhook) getAllLatestLibraries() []libInfo {
 	return libsToInject
 }
 
-// extractLibInfo returns the language, the image,
-// and a boolean indicating whether the library should be injected into the pod
-func (w *Webhook) extractLibInfo(pod *corev1.Pod) ([]libInfo, bool) {
+type libInfoSource int
+
+const (
+	libInfoSourceNone libInfoSource = iota
+	libInfoSourceLibInjection
+	libInfoSourceSingleStepInstrumentation
+	libInfoSourceSingleStepLangaugeDetection
```

Comment:
Can you please explain what each of these refer to? 

I might be able to guess, but want to make sure I understand correctly, and maybe we can improve the naming so that the names convey the intent better.